File: app/src/main/java/com/genonbeta/TrebleShot/view/LongTextBubbleFastScrollViewProvider.java
package com.genonbeta.TrebleShot.view;

import android.graphics.drawable.InsetDrawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.genonbeta.TrebleShot.R;
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.Utils;
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.DefaultBubbleBehavior;
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.ScrollerViewProvider;
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.ViewBehavior;
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.VisibilityAnimationManager;

/**
 * created by: veli
 * date: 10.04.2018 19:52
 */

public class LongTextBubbleFastScrollViewProvider extends ScrollerViewProvider
{
    private View mBubble;
    private View mHandle;

    @Override
    public View provideHandleView(ViewGroup container)
    {
        mHandle = new View(getContext());

        int verticalInset = getScroller().isVertical() ? 0 : getContext().getResources().getDimensionPixelSize(com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_inset);
        int horizontalInset = !getScroller().isVertical() ? 0 : getContext().getResources().getDimensionPixelSize(com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_inset);
        InsetDrawable handleBg = new InsetDrawable(ContextCompat.getDrawable(getContext(), com.genonbeta.android.framework.R.drawable.genfw_fastscroll_default_handle), horizontalInset, verticalInset, horizontalInset, verticalInset);
        Utils.setBackground(mHandle, handleBg);

        int handleWidth = getContext().getResources().getDimensionPixelSize(getScroller().isVertical() ? com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_clickable_width : com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_height);
        int handleHeight = getContext().getResources().getDimensionPixelSize(getScroller().isVertical() ? com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_height : com.genonbeta.android.framework.R.dimen.genfw_fastscroll_handle_clickable_width);
        ViewGroup.LayoutParams params = new ViewGroup.LayoutParams(handleWidth, handleHeight);

        mHandle.setLayoutParams(params);

        return mHandle;
    }

    @Override
    public View provideBubbleView(ViewGroup container)
    {
        mBubble = LayoutInflater.from(getContext()).inflate(R.layout.abstract_layout_fast_scroll_long_text_bubble_text_view, container, false);
        return mBubble;
    }

    @Override
    public TextView provideBubbleTextView()
    {
        return (TextView) mBubble;
    }

    @Override
    public int getBubbleOffset()
    {
        return (int) (getScroller().isVertical() ? (float) mHandle.getHeight() / 2f - (float) mBubble.getHeight() / 2f : (float) mHandle.getWidth() / 2f - (float) mBubble.getWidth() / 2);
    }

    @Override
    protected ViewBehavior provideHandleBehavior()
    {
        return null;
    }

    @Override
    protected ViewBehavior provideBubbleBehavior()
    {
        return new DefaultBubbleBehavior(new VisibilityAnimationManager.Builder(mBubble).withPivotX(1f).withPivotY(1f).build());
    }
}

File: app/src/main/java/com/genonbeta/TrebleShot/view/LongTextBubbleFastScrollViewProvider.kt
/*
 * Copyright (C) 2019 Veli Tasalı
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.genonbeta.TrebleShot.view

import android.graphics.drawable.InsetDrawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import com.genonbeta.TrebleShot.R
import com.genonbeta.android.framework.R.dimen
import com.genonbeta.android.framework.R.drawable
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.Utils
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.DefaultBubbleBehavior
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.ScrollViewProvider
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.ViewBehavior
import com.genonbeta.android.framework.widget.recyclerview.fastscroll.provider.VisibilityAnimationManager

/**
 * created by: veli
 * date: 10.04.2018 19:52
 */
class LongTextBubbleFastScrollViewProvider : ScrollViewProvider() {
    override lateinit var bubbleTextView: TextView

    override lateinit var bubbleView: View

    override lateinit var handleView: View

    override fun getBubbleOffset(): Int = if (scroller.horizontalLayout) {
        (handleView.width.toFloat() / 2f - bubbleView.width.toFloat() / 2).toInt()
    } else {
        (handleView.height.toFloat() / 2f - bubbleView.height.toFloat() / 2f).toInt()
    }

    override fun recreateViews(container: ViewGroup?) {
        bubbleTextView = LayoutInflater.from(context).inflate(
            R.layout.abstract_layout_fast_scroll_long_text_bubble_text_view, container, false
        ) as TextView
        bubbleView = bubbleTextView
        handleView = View(context).also {
            val inset = context.resources.getDimensionPixelSize(dimen.genfw_fastscroll_handle_inset)
            val verticalInset = if (scroller.horizontalLayout) inset else 0
            val horizontalInset = if (scroller.horizontalLayout) 0 else inset
            val handleBackground = InsetDrawable(
                ContextCompat.getDrawable(
                    context,
                    drawable.genfw_fastscroll_default_handle
                ), horizontalInset, verticalInset, horizontalInset, verticalInset
            )

            Utils.setBackground(handleView, handleBackground)

            val handleWidth = context.resources.getDimensionPixelSize(
                if (scroller.horizontalLayout) {
                    dimen.genfw_fastscroll_handle_height
                } else {
                    dimen.genfw_fastscroll_handle_clickable_width
                }
            )
            val handleHeight = context.resources.getDimensionPixelSize(
                if (scroller.horizontalLayout) {
                    dimen.genfw_fastscroll_handle_clickable_width
                } else {
                    dimen.genfw_fastscroll_handle_height
                }
            )

            it.layoutParams = ViewGroup.LayoutParams(handleWidth, handleHeight)
        }

        bubbleBehavior = DefaultBubbleBehavior(VisibilityAnimationManager.Builder(bubbleView)
            .withPivotX(1f)
            .withPivotY(1f)
            .build())
    }
}